e2etests/core/run_sharing_test.go

```diff
@@ -292,29 +267,69 @@ func TestRunSharing(t *testing.T) {
 		},
 	} {
 		tc := tc
-		t.Run(tc.name, func(t *testing.T) {
-			s := sandbox.New(t)
-			s.BuildTree(tc.layout)
-			tmcli := NewCLI(t, s.RootDir())
-			tmcli.PrependToPath(filepath.Dir(TerraformTestPath))
-			res := tmcli.Run("run", HelperPath, "echo", "hello")
-			if res.Status == 0 {
-				// generate safeguard must trigger
-				t.Fatal("run must fail if sharing is not generated")
+		runCases := []bool{false, true}
```

Comment:
*nitpick* This one comes from my ah moment when I saw the var name, no need to apply but I feel it more readable.
```suggestion
		scriptCaseRuns := []bool{false, true}
```

Reply:
I will merge the PR stack and address this and the other request in a separate PR.